src/Portable.Xaml/Portable.Xaml/XamlObjectWriter.cs

```diff
@@ -862,19 +865,23 @@ void ResolvePendingReferences ()
 
 		void HandleBeginInit (object value)
 		{
+#if HAS_ISUPPORT_INITIALIZE
 			var si = value as ISupportInitialize;
 			if (si == null)
 				return;
 			si.BeginInit ();
+#endif
 			source.OnAfterBeginInit (value);
```

Comment:
This line would not have run if the class didn't implement ISupportInitialize before, as it would return above.  It should probably be included in the `#if` block, same with the EndInit

Reply:
Yea you are right. This is my mistake.